src/core/shared_memory_manager.h

```diff
@@ -74,14 +92,15 @@ class SharedMemoryManager {
   /// registered.
   /// \param offset The offset into the shared memory region.
   /// \param byte_size The size, in bytes of the tensor data.
-  /// \parm type The type action to be performed. If the action is REGISTER and
-  /// the shared memory region has been registered, the shared memory region
-  /// will be re-registered.
+  /// \param kind The kind of device the shared memory region is in (CPU = 0,
+  /// GPU = 1)
+  /// \param device id The GPU number the shared memory region is in. Ignored
+  /// if CPU.
   /// \return error status. Return an error if it tries to register a shared
   /// memory region that has already been registered.
   Status RegisterSharedMemory(
       const std::string& name, const std::string& shm_key, const size_t offset,
-      const size_t byte_size);
+      const size_t byte_size, const int kind, const int device_id);
 
   /// Unregister a specified shared memory region if registered else do nothing
   /// and return success.
```

Comment:
No special handling for unregistering CUDA shared memory is required?

Reply:
In the case of CUDA shared memory the `shm_key` the name of the system shared memory region that contains the handle at the `offset`. The byte size here should be the size of the cuda IPC object. 
In case of CUDA shared memory the mapped_addr is used to get the IPC handle which in turn is used to get the device address denoting the start of the CUDA IPC region

Reply:
Hmmm. I was thinking you will need to close a CUDA IPC handle if you opened one, is that no the case?

Reply:
Yes you do need to close the handle after you open it. This is done in the Unregister step now.